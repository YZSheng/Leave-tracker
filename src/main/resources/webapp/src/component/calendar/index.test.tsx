import * as React from 'react';
import { shallow } from 'enzyme';
import Calendar from './index';

describe('Calendar component', () => {
    it('should match snapshot', () => {
        expect(shallow(<Calendar />)).toMatchSnapshot();
    });
});