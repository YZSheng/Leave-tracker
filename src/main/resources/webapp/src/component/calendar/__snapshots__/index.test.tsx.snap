// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Calendar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Uncontrolled(Calendar)
        defaultDate={2018-08-08T16:00:00.000Z}
        events={
          Array [
            Object {
              "allDay": true,
              "end": 2018-07-31T16:00:00.000Z,
              "id": 0,
              "start": 2018-07-30T16:00:00.000Z,
              "title": "All Day Event very long title",
            },
            Object {
              "end": 2018-08-09T16:00:00.000Z,
              "id": 1,
              "start": 2018-08-06T16:00:00.000Z,
              "title": "Long Event",
            },
            Object {
              "end": 2016-03-19T16:00:00.000Z,
              "id": 2,
              "start": 2016-03-12T16:00:00.000Z,
              "title": "DTS STARTS",
            },
            Object {
              "end": 2016-11-12T16:00:00.000Z,
              "id": 3,
              "start": 2016-11-05T16:00:00.000Z,
              "title": "DTS ENDS",
            },
            Object {
              "end": 2018-08-08T16:00:00.000Z,
              "id": 4,
              "start": 2018-08-08T16:00:00.000Z,
              "title": "Some Event",
            },
            Object {
              "desc": "Big conference for important people",
              "end": 2018-08-12T16:00:00.000Z,
              "id": 5,
              "start": 2018-08-10T16:00:00.000Z,
              "title": "Conference",
            },
            Object {
              "desc": "Pre-meeting meeting, to prepare for the meeting",
              "end": 2018-08-12T04:30:00.000Z,
              "id": 6,
              "start": 2018-08-12T02:30:00.000Z,
              "title": "Meeting",
            },
            Object {
              "desc": "Power lunch",
              "end": 2018-08-12T05:00:00.000Z,
              "id": 7,
              "start": 2018-08-12T04:00:00.000Z,
              "title": "Lunch",
            },
            Object {
              "end": 2018-08-12T07:00:00.000Z,
              "id": 8,
              "start": 2018-08-12T06:00:00.000Z,
              "title": "Meeting",
            },
            Object {
              "desc": "Most important meal of the day",
              "end": 2018-08-12T09:30:00.000Z,
              "id": 9,
              "start": 2018-08-12T09:00:00.000Z,
              "title": "Happy Hour",
            },
            Object {
              "end": 2018-08-12T13:00:00.000Z,
              "id": 10,
              "start": 2018-08-12T12:00:00.000Z,
              "title": "Dinner",
            },
            Object {
              "end": 2018-08-13T02:30:00.000Z,
              "id": 11,
              "start": 2018-08-12T23:00:00.000Z,
              "title": "Birthday Party",
            },
            Object {
              "end": 2018-08-17T18:00:00.000Z,
              "id": 12,
              "start": 2018-08-17T11:30:00.000Z,
              "title": "Late Night Event",
            },
            Object {
              "end": 2018-08-21T18:00:00.000Z,
              "id": 13,
              "start": 2018-08-20T11:30:00.000Z,
              "title": "Multi-day Event",
            },
          ]
        }
        showMultiDayTime={true}
        step={60}
        views={
          Array [
            "month",
            "day",
            "agenda",
          ]
        }
      />,
      "className": "calendar-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "defaultDate": 2018-08-08T16:00:00.000Z,
        "events": Array [
          Object {
            "allDay": true,
            "end": 2018-07-31T16:00:00.000Z,
            "id": 0,
            "start": 2018-07-30T16:00:00.000Z,
            "title": "All Day Event very long title",
          },
          Object {
            "end": 2018-08-09T16:00:00.000Z,
            "id": 1,
            "start": 2018-08-06T16:00:00.000Z,
            "title": "Long Event",
          },
          Object {
            "end": 2016-03-19T16:00:00.000Z,
            "id": 2,
            "start": 2016-03-12T16:00:00.000Z,
            "title": "DTS STARTS",
          },
          Object {
            "end": 2016-11-12T16:00:00.000Z,
            "id": 3,
            "start": 2016-11-05T16:00:00.000Z,
            "title": "DTS ENDS",
          },
          Object {
            "end": 2018-08-08T16:00:00.000Z,
            "id": 4,
            "start": 2018-08-08T16:00:00.000Z,
            "title": "Some Event",
          },
          Object {
            "desc": "Big conference for important people",
            "end": 2018-08-12T16:00:00.000Z,
            "id": 5,
            "start": 2018-08-10T16:00:00.000Z,
            "title": "Conference",
          },
          Object {
            "desc": "Pre-meeting meeting, to prepare for the meeting",
            "end": 2018-08-12T04:30:00.000Z,
            "id": 6,
            "start": 2018-08-12T02:30:00.000Z,
            "title": "Meeting",
          },
          Object {
            "desc": "Power lunch",
            "end": 2018-08-12T05:00:00.000Z,
            "id": 7,
            "start": 2018-08-12T04:00:00.000Z,
            "title": "Lunch",
          },
          Object {
            "end": 2018-08-12T07:00:00.000Z,
            "id": 8,
            "start": 2018-08-12T06:00:00.000Z,
            "title": "Meeting",
          },
          Object {
            "desc": "Most important meal of the day",
            "end": 2018-08-12T09:30:00.000Z,
            "id": 9,
            "start": 2018-08-12T09:00:00.000Z,
            "title": "Happy Hour",
          },
          Object {
            "end": 2018-08-12T13:00:00.000Z,
            "id": 10,
            "start": 2018-08-12T12:00:00.000Z,
            "title": "Dinner",
          },
          Object {
            "end": 2018-08-13T02:30:00.000Z,
            "id": 11,
            "start": 2018-08-12T23:00:00.000Z,
            "title": "Birthday Party",
          },
          Object {
            "end": 2018-08-17T18:00:00.000Z,
            "id": 12,
            "start": 2018-08-17T11:30:00.000Z,
            "title": "Late Night Event",
          },
          Object {
            "end": 2018-08-21T18:00:00.000Z,
            "id": 13,
            "start": 2018-08-20T11:30:00.000Z,
            "title": "Multi-day Event",
          },
        ],
        "showMultiDayTime": true,
        "step": 60,
        "views": Array [
          "month",
          "day",
          "agenda",
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Uncontrolled(Calendar)
          defaultDate={2018-08-08T16:00:00.000Z}
          events={
            Array [
              Object {
                "allDay": true,
                "end": 2018-07-31T16:00:00.000Z,
                "id": 0,
                "start": 2018-07-30T16:00:00.000Z,
                "title": "All Day Event very long title",
              },
              Object {
                "end": 2018-08-09T16:00:00.000Z,
                "id": 1,
                "start": 2018-08-06T16:00:00.000Z,
                "title": "Long Event",
              },
              Object {
                "end": 2016-03-19T16:00:00.000Z,
                "id": 2,
                "start": 2016-03-12T16:00:00.000Z,
                "title": "DTS STARTS",
              },
              Object {
                "end": 2016-11-12T16:00:00.000Z,
                "id": 3,
                "start": 2016-11-05T16:00:00.000Z,
                "title": "DTS ENDS",
              },
              Object {
                "end": 2018-08-08T16:00:00.000Z,
                "id": 4,
                "start": 2018-08-08T16:00:00.000Z,
                "title": "Some Event",
              },
              Object {
                "desc": "Big conference for important people",
                "end": 2018-08-12T16:00:00.000Z,
                "id": 5,
                "start": 2018-08-10T16:00:00.000Z,
                "title": "Conference",
              },
              Object {
                "desc": "Pre-meeting meeting, to prepare for the meeting",
                "end": 2018-08-12T04:30:00.000Z,
                "id": 6,
                "start": 2018-08-12T02:30:00.000Z,
                "title": "Meeting",
              },
              Object {
                "desc": "Power lunch",
                "end": 2018-08-12T05:00:00.000Z,
                "id": 7,
                "start": 2018-08-12T04:00:00.000Z,
                "title": "Lunch",
              },
              Object {
                "end": 2018-08-12T07:00:00.000Z,
                "id": 8,
                "start": 2018-08-12T06:00:00.000Z,
                "title": "Meeting",
              },
              Object {
                "desc": "Most important meal of the day",
                "end": 2018-08-12T09:30:00.000Z,
                "id": 9,
                "start": 2018-08-12T09:00:00.000Z,
                "title": "Happy Hour",
              },
              Object {
                "end": 2018-08-12T13:00:00.000Z,
                "id": 10,
                "start": 2018-08-12T12:00:00.000Z,
                "title": "Dinner",
              },
              Object {
                "end": 2018-08-13T02:30:00.000Z,
                "id": 11,
                "start": 2018-08-12T23:00:00.000Z,
                "title": "Birthday Party",
              },
              Object {
                "end": 2018-08-17T18:00:00.000Z,
                "id": 12,
                "start": 2018-08-17T11:30:00.000Z,
                "title": "Late Night Event",
              },
              Object {
                "end": 2018-08-21T18:00:00.000Z,
                "id": 13,
                "start": 2018-08-20T11:30:00.000Z,
                "title": "Multi-day Event",
              },
            ]
          }
          showMultiDayTime={true}
          step={60}
          views={
            Array [
              "month",
              "day",
              "agenda",
            ]
          }
        />,
        "className": "calendar-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultDate": 2018-08-08T16:00:00.000Z,
          "events": Array [
            Object {
              "allDay": true,
              "end": 2018-07-31T16:00:00.000Z,
              "id": 0,
              "start": 2018-07-30T16:00:00.000Z,
              "title": "All Day Event very long title",
            },
            Object {
              "end": 2018-08-09T16:00:00.000Z,
              "id": 1,
              "start": 2018-08-06T16:00:00.000Z,
              "title": "Long Event",
            },
            Object {
              "end": 2016-03-19T16:00:00.000Z,
              "id": 2,
              "start": 2016-03-12T16:00:00.000Z,
              "title": "DTS STARTS",
            },
            Object {
              "end": 2016-11-12T16:00:00.000Z,
              "id": 3,
              "start": 2016-11-05T16:00:00.000Z,
              "title": "DTS ENDS",
            },
            Object {
              "end": 2018-08-08T16:00:00.000Z,
              "id": 4,
              "start": 2018-08-08T16:00:00.000Z,
              "title": "Some Event",
            },
            Object {
              "desc": "Big conference for important people",
              "end": 2018-08-12T16:00:00.000Z,
              "id": 5,
              "start": 2018-08-10T16:00:00.000Z,
              "title": "Conference",
            },
            Object {
              "desc": "Pre-meeting meeting, to prepare for the meeting",
              "end": 2018-08-12T04:30:00.000Z,
              "id": 6,
              "start": 2018-08-12T02:30:00.000Z,
              "title": "Meeting",
            },
            Object {
              "desc": "Power lunch",
              "end": 2018-08-12T05:00:00.000Z,
              "id": 7,
              "start": 2018-08-12T04:00:00.000Z,
              "title": "Lunch",
            },
            Object {
              "end": 2018-08-12T07:00:00.000Z,
              "id": 8,
              "start": 2018-08-12T06:00:00.000Z,
              "title": "Meeting",
            },
            Object {
              "desc": "Most important meal of the day",
              "end": 2018-08-12T09:30:00.000Z,
              "id": 9,
              "start": 2018-08-12T09:00:00.000Z,
              "title": "Happy Hour",
            },
            Object {
              "end": 2018-08-12T13:00:00.000Z,
              "id": 10,
              "start": 2018-08-12T12:00:00.000Z,
              "title": "Dinner",
            },
            Object {
              "end": 2018-08-13T02:30:00.000Z,
              "id": 11,
              "start": 2018-08-12T23:00:00.000Z,
              "title": "Birthday Party",
            },
            Object {
              "end": 2018-08-17T18:00:00.000Z,
              "id": 12,
              "start": 2018-08-17T11:30:00.000Z,
              "title": "Late Night Event",
            },
            Object {
              "end": 2018-08-21T18:00:00.000Z,
              "id": 13,
              "start": 2018-08-20T11:30:00.000Z,
              "title": "Multi-day Event",
            },
          ],
          "showMultiDayTime": true,
          "step": 60,
          "views": Array [
            "month",
            "day",
            "agenda",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
